### ISSUE TITLE
Deny doesn't work if allow is not specified.

### ISSUE THREAD

**Issue by RokLenarcic** [ptaoussanis/encore#55]
```
((enc/compile-str-filter {:deny #{"org.eclipse.jetty.util.thread.*"}})
 "org.eclipse.jetty.util.thread.QueuedThreadPool")
=> true
((enc/compile-str-filter {:allow #{"*"} :deny #{"org.eclipse.jetty.util.thread.*"}})
 "org.eclipse.jetty.util.thread.QueuedThreadPool")
=> false
```

It is because of this code:
```
          (and allow deny)
          (fn [?in-str]
            ...)

          allow (if (= allow always) always (fn [?in-str] (if (allow (str ?in-str)) true false)))
          deny  (if (= deny  never)  always (fn [?in-str] (if (deny  (str ?in-str)) true false)))

```
You'll notice that `fn [?in-str] ... ` is the same for both solo allow and solo deny, which I believe is incorrect.

**Comment by ptaoussanis** [comment]
@RokLenarcic Hi Rok, thanks a lot for the catch! Fixed with `v3.19.0`, now on Clojars.

**Comment by RokLenarcic** [comment]
Btw, this affects `timbre`, that's how I got to this.
